spidermon/contrib/scrapy/pipelines.py

```diff
@@ -109,8 +109,7 @@ def process_item(self, item, _):
             # No validators match this specific item type
             return item
 
-        item_adapter = ItemAdapter(item)
```

Comment:
**issue:** `item_adapter` is needed later for `self._add_errors_to_item(item_adapter, errors)` below.

**suggestion:** Maybe a test case could be added to check that case.

That's probably why `_convert_item_to_dict()` was removed. We needed the item adapter later, which that method discarded otherwise.

I think the removal of `_convert_item_to_dict()` isn't necessarily a backward-incompatible change, as developers shouldn't depend on internal methods. We could restore it, but this pipeline won't use it (as it needs a reference to the adapter object). Or it might use it with some modifications, but at that point, developers will have to refactor their ItemValidationPipeline extensions to accommodate those modifications, too.

@arslansherazi, could you show us some examples of how this pipeline was extended so we can see how to support that case?